src/downloader.c

```diff
@@ -1597,11 +1600,8 @@ static void recover_shards(uv_work_t *work)
 
 static void queue_recover_shards(storj_download_state_t *state)
 {
-    if (state->recovering_shards) {
-        return;
-    }
-
-    if (state->pointers_completed) {
+    if (!state->recovering_shards && state->pointers_completed) {
+        state->recovering_shards = true;
 
         int total_missing = 0;
         bool has_missing = false;
```

Comment:
On line 1626: there is a return that doesn't set `state->recovering_shards` to `false`, and it would remain `true` even though it's not queued, and thus may never get queued.